helpers/helpers.rb

```diff
@@ -30,12 +30,15 @@ def get_timeline_view(user)
     end
 
     def get_global_timeline
-      if $redis.exists("timeline:recent:50")
+      if $redis.ttl("timeline:recent:50")>0
         @tweets = $redis.lrange("timeline:recent:50", 0, -1).map { |t| Tweet.new(JSON.parse(t)) }
       else
         @tweets = Tweet.all.order(tweeted_at: :desc).take(50)
         @timeline = @tweets.map { |t| t.to_json }
-        $redis.rpush("timeline:recent:50", @timeline) if !@timeline.empty?
+        if !@timeline.empty?
+          $redis.rpush("timeline:recent:50", @timeline)
+          $redis.expire("timeline:recent:50",30)
```

Comment:
I think this doesn't need to expire since we are updating global timeline immediately after a new tweet is posted.